libs/sqf/src/analyze/lints/s30_configof.rs

```diff
@@ -0,0 +1,141 @@
+use std::{ops::Range, sync::Arc};
+
+use hemtt_common::config::LintConfig;
+use hemtt_workspace::{
+    lint::{AnyLintRunner, Lint, LintRunner},
+    reporting::{Code, Codes, Diagnostic, Processed, Severity},
+};
+
+use crate::{analyze::LintData, BinaryCommand, Expression, NularCommand, UnaryCommand};
+
+crate::analyze::lint!(LintS30ConfigOf);
+
+impl Lint<LintData> for LintS30ConfigOf {
+    fn ident(&self) -> &'static str {
+        "config_of"
+    }
+
+    fn sort(&self) -> u32 {
+        300
+    }
+
+    fn description(&self) -> &'static str {
+        "Checks for typeOf used with configFile when configOf could be used instead"
+    }
+
+    fn documentation(&self) -> &'static str {
+        r"### Example
+
+**Incorrect**
+```sqf
+private _name = getText(configFile >> 'CfgVehicles' >> typeOf _vehicle >> 'displayName');
+```
+**Correct**
+```sqf
+private _name = getText(configOf _vehicle >> 'displayName');
+```
+
+### Explanation
+
+The `configOf` command is specifically designed to retrieve configuration data for a given object, and is faster and more efficient than using `typeOf` in conjunction with `configFile`.
+"
+    }
+
+    fn default_config(&self) -> LintConfig {
+        LintConfig::warning()
+    }
+
+    fn runners(&self) -> Vec<Box<dyn AnyLintRunner<LintData>>> {
+        vec![Box::new(Runner)]
+    }
+}
+
+struct Runner;
+impl LintRunner<LintData> for Runner {
+    type Target = crate::Expression;
+
+    fn run(
+        &self,
+        _project: Option<&hemtt_common::config::ProjectConfig>,
+        config: &LintConfig,
+        processed: Option<&hemtt_workspace::reporting::Processed>,
+        _runtime: &hemtt_common::config::RuntimeArguments,
+        target: &Self::Target,
+        _data: &LintData,
+    ) -> Codes {
+        let Some(processed) = processed else {
+            return Vec::new();
+        };
+        let Expression::BinaryCommand(BinaryCommand::ConfigPath, lhs, rhs, _) = target else {
+            return Vec::new();
+        };
+
+        if let Expression::UnaryCommand(UnaryCommand::Named(rhs_cmd), rhs_rhs, _ ) = &**rhs
+        && rhs_cmd.eq_ignore_ascii_case("typeof")
+        && let Expression::BinaryCommand(BinaryCommand::ConfigPath, lhs_lhs, lhs_rhs, _) = &**lhs
```

Comment:
technically someone could be using
`configFile / "CfgVehicles" / typeOf player;`
but I doubt it's needed